README.md
# simple-assert

> Vanilla Assertions

A simple `assert` wrapper around [chaijs/assertion-error](https://github.com/chaijs/assertion-error). This probably
won't be useful to the average user unless you are a minimalist; you probably want [Chai](https://github.com/chaijs/chai).
We use it to avoid circular dependencies when testing Chai's dependencies.

## Installation

### Node.js

`simple-assert` is available on [npm](http://npmjs.org).

    $ npm install simple-assert

### Component

`simple-assert` is available as a [component](https://github.com/component/component).

    $ component install chaijs/simple-assert

## Usage

### assert (expr[, msg])

* **@param** _{Mixed}_ expression to test for truthiness
* **@param** _{String}_ message on failure

Perform a truthy assertion.

```js
var assert = require('simple-assert');
assert(true, 'true is truthy');
assert(1, '1 is truthy');
assert('string', 'string is truthy');
```


### assert.not (expr[, msg])

* **@param** _{Mixed}_ express to test for falsiness
* **@param** _{String}_ messag eon failure

Perform a falsey assertion.

```js
db.get(123, function (err, doc) {
  assert.not(err, 'db.get returned error');
  // ...
});
```


### assert.fail ([msg])

* **@param** _{String}_ failure message

Force an `AssertionError` to be thrown.

```js
switch (res.statusCode) {
  case 200:
    // ..
    break;
  case 404:
    // ..
    break;
  default:
    assert.fail('Unknown response statusCode');
}
```

## License

(The MIT License)

Copyright (c) 2012 Jake Luer <jake@alogicalparadox.com> (http://alogicalparadox.com)

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.


CHANGELOG.md
## 2.0.0

* Convert project to an ES module

## 1.0.0

* makefile: cleanup
* add node 0.10
* cleaning up:
* readme
* add tests
* lib: export assert function
* deps: add assertion-error
* "Initial commit"


test.js
import {AssertionError} from 'assertion-error'
import {assert, fail, not} from './dist/bundled.js'

function err (fn, msg) {
  var thrown = false;
  try {
    fn();
  } catch (ex) {
    thrown = true;

    if (msg && msg !== ex.message) {
      var report = 'Expected "' + ex.message + '" to equal "' + msg + '".'
      throw new AssertionError(report);
    }
  }

  if (!thrown) {
    throw new AssertionError('Expected an error.');
  }
}

it('assert(expr)', function () {
  assert('string', 'string');

  assert(true, 'true');
  err(function () {
    assert(false, 'test message');
  }, 'test message');

  assert(1, '1');
  err(function () {
    assert(0, 'test message');
  }, 'test message');

  err(function () {
    assert(null, 'test message');
  }, 'test message');

  err(function () {
    assert(undefined, 'test message');
  }, 'test message');
});

it('not(expr)', function () {
  not(false, 'false');
  err(function () {
    not(true, 'test message');
  }, 'test message');

  not(0, '0');
  err(function () {
    not(1, 'test message');
  }, 'test message');

  not(null, 'null');
  not(undefined, 'undefined');
  err(function () {
    not('string', 'test message');
  }, 'test message');
});

it('fail(msg)', function () {
  err(function () {
    fail('test message');
  }, 'test message');
});


test/assert.js
import {AssertionError} from 'assertion-error'
import {assert, fail, not} from '../dist/bundled.js'

function err (fn, msg) {
  var thrown = false;
  try {
    fn();
  } catch (ex) {
    thrown = true;

    if (msg && msg !== ex.message) {
      var report = 'Expected "' + ex.message + '" to equal "' + msg + '".'
      throw new AssertionError(report);
    }
  }

  if (!thrown) {
    throw new AssertionError('Expected an error.');
  }
}

it('assert(expr)', function () {
  assert('string', 'string');

  assert(true, 'true');
  err(function () {
    assert(false, 'test message');
  }, 'test message');

  assert(1, '1');
  err(function () {
    assert(0, 'test message');
  }, 'test message');

  err(function () {
    assert(null, 'test message');
  }, 'test message');

  err(function () {
    assert(undefined, 'test message');
  }, 'test message');
});

it('not(expr)', function () {
  not(false, 'false');
  err(function () {
    not(true, 'test message');
  }, 'test message');

  not(0, '0');
  err(function () {
    not(1, 'test message');
  }, 'test message');

  not(null, 'null');
  not(undefined, 'undefined');
  err(function () {
    not('string', 'test message');
  }, 'test message');
});

it('fail(msg)', function () {
  err(function () {
    fail('test message');
  }, 'test message');
});


test/bootstrap/index.js
/*!
 * Attach simple-assert to global assert
 */

global.assert = require('../..');


index.ts
import {AssertionError} from 'assertion-error'

/**
 * ### assert (expr[, msg])
 *
 * Perform a truthy assertion.
 *
 * ```js
 * var assert = require('simple-assert');
 * assert(true, 'true is truthy');
 * assert(1, '1 is truthy');
 * assert('string', 'string is truthy');
 * ```
 *
 * @throws AssertionError
 */

export function assert (expr: unknown, msg: string) {
  if (!expr) {
    throw new AssertionError(msg || 'Assertion Failed');
  }
}

/**
 * ### assert.not (expr[, msg])
 *
 * Perform a falsey assertion.
 *
 * ```js
 * db.get(123, function (err, doc) {
 *   assert.not(err, 'db.get returned error');
 *   // ...
 * });
 * ```
 *
 * @throws AssertionError
 */

export function not (expr: unknown, msg: string) {
  assert(!expr, msg);
};

/**
 * ### assert.fail ([msg])
 *
 * Force an `AssertionError` to be thrown.
 *
 * ```js
 * switch (res.statusCode) {
 *   case 200:
 *     // ..
 *     break;
 *   case 404:
 *     // ..
 *     break;
 *   default:
 *     assert.fail('Unknown response statusCode');
 * }
 * ```
 *
 * @throws AssertionError
 */

export function fail(msg: string) {
  assert(false, msg);
};
